pkg/reconciler/revision/resources/deploy.go

```diff
@@ -136,49 +136,60 @@ func makePodSpec(rev *v1.Revision, loggingConfig *logging.Config, tracingConfig
 	return podSpec, nil
 }
 
-// BuildUserContainer makes a container from the Revision template.
-func BuildUserContainer(rev *v1.Revision) *corev1.Container {
-	userContainer := rev.Spec.GetContainer().DeepCopy()
+// BuildUserContainers makes a container from the Revision template.
+func BuildUserContainers(rev *v1.Revision) []corev1.Container {
+	containers := make([]corev1.Container, 0, len(rev.Spec.PodSpec.Containers))
+	for i := range rev.Spec.PodSpec.Containers {
+		var container corev1.Container
+		if len(rev.Spec.PodSpec.Containers[i].Ports) != 0 || len(rev.Spec.PodSpec.Containers) == 1 {
+			container = makeServingContainer(*rev.Spec.GetContainer(), rev)
```

Comment:
Why not just pass `rev.Spec.PodSpec.Containers[i]` in this case?

Reply:
We can pass 
we already had `GetContainer()` function so used that only

Reply:
Yeah let's be consistent here.